apps/src/block_utils.js

```diff
@@ -442,17 +466,16 @@ const FIELD_INPUT = 'field';
  * @returns {LabeledInputConfig[]} a list of labeled inputs
  */
 const determineInputs = function (text, args, strictTypes=[]) {
-  const tokens = text.split(/[{}]/);
-  if (tokens[tokens.length - 1] === '') {
+  const tokens = text.match(LABELED_INPUTS_REGEX);
+  if (tokens.length && tokens[tokens.length - 1] === '') {
     tokens.pop();
   }
-  const inputs = [];
-  for (let i = 0; i < tokens.length; i += 2) {
-    const label = tokens[i];
-    const input = tokens[i + 1];
-    if (input) {
-      const argIndex = args.findIndex(arg => arg.name === input);
-      const [arg] = args.splice(argIndex, 1);
+  const inputs = tokens.map(token => {
+    const parts = token.match(LABELED_INPUT_PARTS_REGEX);
+    const label = parts[1];
+    const inputName = parts[3];
```

Comment:
Not at all blocking, but this makes me wish we had [the Babel plugin for named capture groups](https://github.com/tc39/proposal-regexp-named-groups).